test/view/generate_view.cpp

```diff
@@ -0,0 +1,132 @@
+#include <stl2/view/generate_view.hpp>
+
+#include "../simple_test.hpp"
+#include <cstring>
+#include <stl2/ranges.hpp>
+#include <stl2/view/take_exactly.hpp>
+
+namespace ranges = __stl2;
+
+namespace view {
+	using namespace ranges::view;
+	using namespace ext;
+} // namespace view
+
+// a simple type to test move semantics
+struct MoveOnlyString {
+	char const* sz_;
+
+	MoveOnlyString(char const* sz = "")
+	  : sz_(sz)
+	{}
+	MoveOnlyString(MoveOnlyString&& that)
+	  : sz_(that.sz_)
+	{
+		that.sz_ = "";
+	}
+	MoveOnlyString(MoveOnlyString const&) = delete;
+	MoveOnlyString& operator=(MoveOnlyString&& that)
+	{
+		sz_ = that.sz_;
```

Comment:
```suggestion
		sz_ = ranges::exchange(that.sz_, "");
```

Reply:
FYI this will simplify a lot of my move constructors from now on!

Reply:
Yeah, I use `std::exchange(foo, nullptr)` *all* the time.